components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/listener/OAuthCacheRemoveListenerTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.listener;
+
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth.cache.CacheEntry;
+import org.wso2.carbon.identity.oauth.cache.OAuthCache;
+import org.wso2.carbon.identity.oauth.cache.OAuthCacheKey;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+
+import javax.cache.Cache;
+import javax.cache.event.CacheEntryEvent;
+import java.sql.Timestamp;
+import java.util.Calendar;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertEquals;
+
+@PrepareForTest({IdentityUtil.class, OAuthServerConfiguration.class, OAuthCache.class})
+public class OAuthCacheRemoveListenerTest extends PowerMockTestCase {
+
+    @Mock
+    IdentityUtil mockedIdentityUtil;
+
+    @Mock
+    OAuthServerConfiguration oauthServerConfigurationMock;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        initMocks(this);
+        mockStatic(IdentityUtil.class);
+        mockStatic(OAuthServerConfiguration.class);
+        mockStatic(OAuthCache.class);
```

Comment:
Why did you have to mock this?

Reply:
Otherwise, carbon context throws some exception